crates/revm/src/optimism.rs

```diff
@@ -83,19 +77,16 @@ impl L1BlockInfo {
     }
 
     /// Calculate the gas cost of a transaction based on L1 block data posted on L2
-    pub fn calculate_tx_l1_cost<SPEC: Spec>(&self, input: &Bytes, is_deposit: bool) -> U256 {
-        let rollup_data_gas_cost = self.data_gas::<SPEC>(input);
-
-        if is_deposit || rollup_data_gas_cost == U256::ZERO {
-            return U256::ZERO;
-        }
+    pub fn calculate_tx_l1_cost<SPEC: Spec>(&self, input: &Bytes) -> U256 {
+        // input must not be an deposit transaction
+        debug_assert!(!input.is_empty() && input[0] != 0x7E);
```

Comment:
Can we just return `U256::ZERO` if either of these two conditions are met? This makes the function more abstract as opposed to it only being used for non-deposit, non-empty transactions.